sympy/solvers/ode/single.py

```diff
@@ -78,10 +78,14 @@ class SingleODEProblem:
     _eq_high_order_free = None
 
     def __init__(self, eq, func, sym, prep=True, **kwargs):
-        assert isinstance(eq, Expr)
-        assert isinstance(func, AppliedUndef)
-        assert isinstance(sym, Symbol)
-        assert isinstance(prep, bool)
+        if not isinstance(eq, Expr):
+            raise TypeError("The equation 'eq' must be an instance of Expr")
+        if not isinstance(func, AppliedUndef):
+            raise TypeError("The function 'func' must be an instance of AppliedUndef")
+        if not isinstance(sym, Symbol):
+            raise TypeError("The variable 'sym' must be an instance of Symbol")
+        if not isinstance(prep, bool):
+            raise TypeError("Expected 'prep' to be a bool")
```

Comment:
This is an internal class. I think it should be fine to use assert for this because it is validating something that should already have been satisfied by the calling code.

What would be best though is to use type annotations and let a type checker verify that the arguments are of the correct type.